app/models/creatorship.rb

```diff
@@ -4,6 +4,19 @@ class Creatorship < ActiveRecord::Base
 
   before_destroy :expire_caches
 
+  validate :unique_index
```

Comment:
This code looks good, but we were wondering why it's necessary -- is there something specific that was causing creatorships to be invalid due to a non-unique index?